sw/device/tests/kmac_idle_test.c

```diff
@@ -0,0 +1,125 @@
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_clkmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR), &clkmgr));
+
+  // Get initial hint and enable for KMAC clock and check both are enabled.
+  dif_toggle_t clock_hint_state;
+  dif_toggle_t clock_state;
+
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_get_hint(&clkmgr, kmac_clock,
+                                                  &clock_hint_state));
+  CHECK_DIF_OK(
+      dif_clkmgr_hintable_clock_get_enabled(&clkmgr, kmac_clock, &clock_state));
+  CHECK((clock_state == kDifToggleEnabled) &&
+        (clock_hint_state == kDifToggleEnabled));
+
+  // While KMAC is not in use, set disabled hint for KMAC clock
+  // then check clock state, set back to enabled and check again.
+  // Note: disabled means the clock can be disabled.
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, kmac_clock,
+                                                  kDifToggleDisabled));
+  check_clock_state(kDifToggleDisabled);
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, kmac_clock,
+                                                  kDifToggleEnabled));
+  check_clock_state(kDifToggleEnabled);
+
+  // Initialize KMAC hardware.
+  dif_kmac_operation_state_t kmac_operation_state;
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Configure KMAC hardware using software entropy.
+  dif_kmac_config_t config = (dif_kmac_config_t){
+      .entropy_mode = kDifKmacEntropyModeSoftware,
+      .entropy_seed = 0xffff,
+      .entropy_fast_process = kDifToggleEnabled,
+  };
+  CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
+
+  // Run SHA3 test case using single blocking absorb/squeeze operations.
+  CHECK_DIF_OK(dif_kmac_mode_sha3_start(&kmac, &kmac_operation_state,
+                                        sha3_256_test.mode));
+
+  // Set hint and check clk state remains enabled as KMAC is now not idle.
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, kmac_clock,
+                                                  kDifToggleDisabled));
+  check_clock_state(kDifToggleEnabled);
+
+  CHECK_DIF_OK(dif_kmac_absorb(&kmac, &kmac_operation_state,
+                               sha3_256_test.message, sha3_256_test.message_len,
+                               NULL));
+
+  uint32_t out[DIGEST_LEN_SHA3_MAX];
+  CHECK_DIF_OK(dif_kmac_squeeze(&kmac, &kmac_operation_state, out,
+                                sha3_256_test.digest_len, NULL));
+
+  // Check clock state again which should still be enabled.
+  check_clock_state(kDifToggleEnabled);
+
+  CHECK_DIF_OK(dif_kmac_end(&kmac, &kmac_operation_state));
+
+  // Check the clock is now stopped.
+  check_clock_state(kDifToggleDisabled);
+
+  // Check the result to be sure the SHA3 operation completed correctly.
+  CHECK_BUFFER(out, sha3_256_test.digest, sha3_256_test.digest_len,
+               "Digest mismatch for test SHA3 256.");
+
```

Comment:
Maybe redundant, but doesn't hurt to add these steps again, to verify KMAC clock can be enabled again. 
```
  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, kmac_clock,
                                                  kDifToggleEnabled));
  check_clock_state(kDifToggleEnabled);
```

Or, better yet, factor out the test in a function and call it twice. That should prove that we can turn KMAC clock on / off without affecting its operability in any way.